lib/composer.js

```diff
@@ -194,6 +194,10 @@ export default class Composer extends Disposable {
 
     await Promise.all(jobs)
 
+    if (this.shouldCleanAfterBuild()) {
```

Comment:
Chucking this onto a logger feels like a hack. A logging facility shouldn't be responsible for keeping track of build success/error state. If this is currently the best way of solving this, we're going to have to do some refactoring.